token/parse.go

```diff
@@ -76,6 +76,8 @@ func (p Payload) Type() Type {
 		return JWK
 	case p.Email != "":
 		return OIDC
+	case len(p.Audience) > 0 && p.Issuer != "" && p.Subject != "" && !p.Expiry.Time().IsZero() && !p.IssuedAt.Time().IsZero():
+		return OIDC
```

Comment:
Any well-formed token will be like this, wouldn't I make more sense to look for `kubernetes.io`?

Reply:
That would work but what if there are other issuers besides Kubernetes that don't use email in the claims? With this being the last check before interpreting the type as unknown would there be any drawbacks to saying that if it's a valid OIDC token then it gets the OIDC type?

Reply:
Sure, we could even remove the case p.Email != "" and just use that, those fields are required per OIDC spec.

Reply:
948d88c4